website/blog/2020-08-04-prestodb-and-hudi.md

```diff
@@ -0,0 +1,153 @@
+
+**[Support Incremental and point in time time-travel queries on Hudi tables](https://issues.apache.org/jira/browse/HUDI-887)**
+
+Incremental queries allow us to extract change logs from a source Hudi table. Point in time queries allows for getting the state of a hudi table between time T1 and T2. These are supported in Hive and Spark already. We are looking into supporting this feature in Presto as well. 
+
+In Hive incremental queries are supported by setting few configs in `JobConf` like for example - query mode to `INCREMENTAL`, starting commit time and max number of commits to consume. In Spark, there is a specific implementation to support incremental querying - `IncrementalRelation`. To support this in Presto, we need a way to identify the incremental query. Given Presto does not pass arbitrary session configs to the hadoop configuration object, an initial idea is to register the same table in the metastore as an incremental table. And then use query predicates to get other details such as starting commit time, max commits etc.
+
+**[RFC-15: Query planning and listing improvements](https://cwiki.apache.org/confluence/display/HUDI/RFC+-+15%3A+HUDI+File+Listing+and+Query+Planning+Improvements)**
+ 
+Hudi  write client and Hudi queries need to perform `listStatus` operation on the file system to get a current view of the file system. While at Uber, the HDFS infrastructure was [heavily optimized](https://eng.uber.com/scaling-hdfs/) for listing, this can be an expensive op for large datasets containing thousands of partitions and each partition having thousands of files on cloud/object storage. The above RFC work aims at eliminating list operation and providing better query performance and faster lookups, by simply incrementally compacting Hudi’s timeline metadata into a snapshot of a table’s state at that instant.
+
+The solutions here aim at 
+- ways for storing and maintaining metadata of the latest list of files 
+- maintaining stats on all columns of a table to aid effective pruning of files before scanning. This can be leveraged in the query planning phase of the engine.
+
+
+Towards this, Presto would need some changes too. We are actively exploring ways to leverage such metadata into the query planning phase. This would be a significant addition to Presto-Hudi integration and would push the query latencies further down.
+
+**[Record Level Indexes](https://cwiki.apache.org/confluence/display/HUDI/RFC+-+08+%3A+Record+level+indexing+mechanisms+for+Hudi+datasets)**
+
+Upsert is a popular write operation on hudi tables that relies on indexing to tag incoming records as upserts. HoodieIndex provides a mapping of record id to a file id in both a partitioned or a non partitioned dataset, with implementations backed by BloomFilters/ Key ranges (for temporal data) and also Apache HBase (for random updates). Many users find Apache HBase (or any such key-value store backed index) expensive and adds to operational overhead. 
```

Comment:
Capitalize Hudi
of record id -> of a record id
or a non partitioned -> or non-partitioned
and also Apache -> , and also Apache

expensive and adds to operational overhead -> expensive and adding to operational overhead